File: 프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.py
from collections import Counter
def solution(array):
    cnt = Counter(array)
    answer = cnt.most_common(2)[0][0]
    
    if (len(cnt.most_common(2))==2):
        if (cnt.most_common(2)[0][1] == cnt.most_common(2)[1][1]):
            answer = -1
    
    return answer

File: Java/프로그래머스/0/120812. 최빈값 구하기/최빈값 구하기.java
import java.util.HashSet;
import java.util.Arrays;
import java.util.Collections;

class Solution {
    public int solution(int[] array) {
        HashSet<Integer> unique = new HashSet<>();
        int answer = 0;
        int maxCnt = 0;
        
        for (int num : array){
            unique.add(num);
        }
        
        for (Integer uniqueNum : unique){
            long cnt = Arrays.stream(array)
                            .filter(num -> num == uniqueNum)
                            .count();
            if (maxCnt == cnt){
                answer = -1;
            } else if (maxCnt < cnt){
                answer = uniqueNum;
                maxCnt = (int)cnt;
            }
        }
        return answer;
    }
}